From e7d45afc6aeb49745f17d21ddba4848e0c0118fc Mon Sep 17 00:00:00 2001
From: Explorer09 <explorer09@gmail.com>
Date: Mon, 13 Nov 2017 02:11:10 +0800
Subject: [PATCH] scanner: Allow "--" as option argument.

Previously, invocations such as:

    flex --outfile -- wc1.l

 would fail with "requires an
argument" error message.

Fixes #285
---
 src/scanopt.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/scanopt.c b/src/scanopt.c
index 04fbdc39..7a91178b 100644
--- a/src/scanopt.c
+++ b/src/scanopt.c
@@ -708,8 +708,7 @@ int     scanopt (scanopt_t *svoid, char **arg, int *optindex)
 
 	/* Look ahead in argv[] to see if there is something
 	 * that we can use as an argument (if needed). */
-	has_next = s->index + 1 < s->argc
-		&& strcmp ("--", s->argv[s->index + 1]) != 0;
+	has_next = s->index + 1 < s->argc;
 
 	optp = s->options + opt_offset;
 	auxp = s->aux + opt_offset;
